src/main/java/sh/okx/railswitch/DetectorRailActivateListener.java

```diff
@@ -0,0 +1,163 @@
+            return;
+        }
+        
+        // Check that the sign has a valid switch type
+        String[] lines = ((Sign) above.getState()).getLines();
+        SwitchType type = SwitchType.find(lines[0]);
+        if (type == null) {
+            return;
+        }
+        
+        // If Citadel is enabled, check that the sign and the rail are on the same group
+        if (Bukkit.getPluginManager().isPluginEnabled("Citadel")) {
+            ReinforcementManager rein = Citadel.getReinforcementManager();
+            Reinforcement railReinforcement = rein.getReinforcement(block);
+            Reinforcement signReinforcement = rein.getReinforcement(above);
+            // Allow the switch to work if the rail and the sign are BOTH unreinforced
+            // otherwise servers with established rail networks will suddenly not work
+            // if Citadel is added.
+            if (railReinforcement != null || signReinforcement != null) {
+                if (!(railReinforcement instanceof PlayerReinforcement)) {
+                    return;
+                }
+                if (!(signReinforcement instanceof PlayerReinforcement)) {
+                    return;
+                }
+                if (!Objects.equals(
+                        ((PlayerReinforcement) railReinforcement).getGroup(),
+                        ((PlayerReinforcement) signReinforcement).getGroup())) {
+                    return;
+                }
+            }
+        }
+        
+        // Check that a player is triggering the switch
+        // NOTE: The event doesn't provide the information and so the next best thing is searching for a
+        //       player who is nearby and riding a minecart.
+        Player player = null; {
+            double searchDistance = Double.MAX_VALUE;
+            for (Entity entity : block.getWorld().getNearbyEntities(block.getLocation(), 3, 3, 3)) {
+                if (entity.getType() != EntityType.PLAYER
+                        || !(entity instanceof Player)) {
+                    continue;
+                }
+                Entity vehicle = entity.getVehicle();
+                if (vehicle == null
+                        || vehicle.getType() != EntityType.MINECART
+                        || !(vehicle instanceof Minecart)) {
+                    continue;
+                }
+                double distance = block.getLocation().distanceSquared(entity.getLocation());
+                if (distance < searchDistance) {
+                    searchDistance = distance;
+                    player = (Player) entity;
+                }
+            }
+        }
+        if (player == null) {
+            return;
+        }
+        
+        // Determine whether a player has a destination that matches one of the destinations
+        // listed on the switch signs, or match if there's a wildcard.
+        boolean matched = false;
+        String[] playerDestinations = plugin.getDatabase().
```

Comment:
Getting rid of this database stuff and using a CMC player setting instead would be desireable, but if you dont feel like doing that its fine.

Reply:
That would be good, but that's a 1.14 CMC feature